src/storm-textarea.js

```diff
@@ -1,37 +1,28 @@
-
 const defaults = {
-		paddingBottom: 16,
-		events: [
-			'focus',
-			'cut',
-			'drop',
-			'keydown',
-			'resize',
-			'change'
-		]
-	},
-	StormTextarea = {
-		init() { 
-			this.settings.events.forEach(evt => this.DOMElement.addEventListener(evt, this.resize.bind(this)));
-			return this;
-		},
-		resize() {
-			this.DOMElement.style.height = 'auto';
-			this.DOMElement.style.height =  this.DOMElement.scrollHeight + this.settings.paddingBottom + 'px';
-		}
-	};
-	
-const init = (sel, opts) => {
-	let els = [].slice.call(document.querySelectorAll(sel));
-	
-	if(els.length === 0) throw new Error('Textarea cannot be initialised, no augmentable elements found');
-	
-	return els.map(el => {
-		return Object.assign(Object.create(StormTextarea), {
-			DOMElement: el,
-			settings: Object.assign({}, defaults, opts)
-		}).init();
-	});
+	events: [
+		'input'
+	]
 };
 
-export default { init };
\ No newline at end of file
+function update({target}) {
+	target.style.height = 'auto';
+	target.style.height =  target.scrollHeight + 'px';
+}
+
+export default function (elements, settings) {
+	if (typeof elements === 'string') {
+		elements = document.querySelectorAll(elements);
+	}
+
+	if(elements.length === 0) {
+		return;
+	}
+
+	const events = settings && settings.events || defaults.events;
+	for (let i = 0; i < elements.length; i++) {
+		for (let l = 0; l < events.length; i++) {
+			elements[i].addEventListener(events[l], update);
+		}
+		update({target: element});
+	}
```

Comment:
This can simply be:

```js
for (const element of elements) {
	for (const event of events) {
		element.addEventListener(event, update);
	}
	update({target: element});
}
```